### PR TITLE
test: Added addFile(), to allow for the creation of files for specific modules for testing purposes

### PR DESCRIPTION
**Pull request by jsamar17**

### CODE REVIEW

File: test/util.ts [hunk 1]
```diff
@@ -16,17 +16,27 @@
...
+const mkdirpP = pify((mkdirp as any).mkdirp);
 const writeFileP = pify(fs.writeFile) as typeof fs.promises.writeFile;
 
 interface PackageJson {
   name: string;
   version: string;
   dependencies: {[key: string]: string;};
+  containedFiles: File[];
+}
+interface File {
```
Comment (kjin) [inline test/util.ts]: Nit: Add newline to separate interfaces and also class methods (later).

Also, can you make this interface more descriptive? Like `FileDetails` or something.

File: test/util.ts [hunk 2]
```diff
@@ -158,8 +170,40 @@ export class TestProject {
             },
             null, 2));
     await execa('npm', ['install'], {cwd: path});
+
```
Comment (kjin) [inline test/util.ts]: Can you remove empty newlines here?

File: test/util.ts [hunk 3]
```diff
@@ -158,8 +170,40 @@ export class TestProject {
...
+    }));
+  }
+  /**
+   * Searches for a module in packageJsons then adds a new file
+   * object to its containedFiles array
+   * @param module The name of the module
+   * @param filePath The path where this file should be located
+   * @param fileContents The contents of the file being written
+   */
+  addFile(module: string, filePath: string, fileContents: string) {
```
Comment (kjin) [inline test/util.ts]: What if there are two versions of the same module?

**Comment by kjin** [comment]
Also, thanks for making the individual commits follow conventional commit guidelines -- that being said, I believe the overall prefix should be `test:`.

File: test/util.ts (hunk 3 above)
Comment (jsamar17) [reply test/util.ts]: I added a version parameter to the method. Now the version and name of the module have to match in order for a file to be added.

File: test/util.ts [hunk 4]
```diff
@@ -170,30 +171,34 @@ export class TestProject {
...
    * @param module The name of the module
+   * @param version The version of the module
+   * Both the version and name of the module have to match, otherwise an error
+   * will be thrown
    * @param filePath The path where this file should be located
    * @param fileContents The contents of the file being written
    */
-  addFile(module: string, filePath: string, fileContents: string) {
+  addFile(
+      module: string, version: string, filePath: string, fileContents: string) {
```
Comment (kjin) [inline test/util.ts]: There's an existing convention where we just specify both `module` and `version` in the same string as `module@version`. Would it be good to uphold that convention here?

Comment (jsamar17) [reply test/util.ts]: Yes, I think so. I'll implement it right now.

**Review by kjin — APPROVED** [review]

**Pull request merged**